source/particle/integrator/rk_4.cc

```diff
@@ -74,107 +74,112 @@ namespace aspect
         typename std::vector<Tensor<1,dim>>::const_iterator old_velocity = old_velocities.begin();
         typename std::vector<Tensor<1,dim>>::const_iterator velocity = velocities.begin();
 
-        std::array<Point<dim>,5> k;
+        Point<dim> old_location;
+        std::array<Tensor<1,dim>,4> k;
         for (typename ParticleHandler<dim>::particle_iterator it = begin_particle;
              it != end_particle; ++it, ++velocity, ++old_velocity)
           {
             ArrayView<double> properties = it->get_properties();
 
             if (integrator_substep == 0)
               {
-                k[0] = it->get_location();
-                k[1] = dt * (*old_velocity);
+                old_location = it->get_location();
+                k[0] = dt * (*old_velocity);
 
-                Point<dim> new_location = k[0] + 0.5 * k[1];
+                Point<dim> new_location = old_location + 0.5 * k[0];
 
                 // Check if we crossed a periodic boundary and if necessary adjust positions
                 if (geometry_has_periodic_boundary)
                   this->get_geometry_model().adjust_positions_for_periodicity(new_location,
-                                                                              ArrayView<Point<dim>>(&k[0],2));
+                                                                              ArrayView<Point<dim>>(&old_location,1),
+                                                                              ArrayView<Tensor<1,dim>>(&k[0],1));
```

Comment:
I think you can get all of this via
```suggestion
                                                                              make_array_view(old_location),
                                                                              make_array_view(k[0]));
```

Reply:
That seems to give me ArrayView<double>, which is not what I need for the other function. I would prefer to keep the type.

Reply:
Oh, I see. Then I meant `make_array_view(k)`, which creates an array view for an array with one element of the type of the element you pass. But you are right that it does not matter.